### ISSUE TITLE
Don’t remove custom files from Gist

### ISSUE THREAD

**Issue by depascalis** [shanalikhan/code-settings-sync#1041]
If uploading to a Gist when the `customFiles` setting is empty, any existing custom files in the Gist will be removed. This is not always the desired result.

I believe I have a simple solution that would also close other issues and requests.

I noticed that you could name a gist “file” to whatever you want, and it doesn’t need to match the actual file name. I believe this opens up for some more advanced configurations.
For example, below there are two machines, one at home and one at work. There are a total of four files that needs to be either synced or just backed up.

_**file1-work**:_ Should only be synced between Gist and WORK
_**file1-home:**_ Another file that shares the file name of above but has other content. Should only be synced between Gist and HOME
_**file2-everywhere:**_ Should be synced with all machines
_**file3:**_ Should only be synced between Gist and WORK

_syncLocalSettings.json at **WORK**_
```
"customFiles": {
    "file1-work": "C:/Users/WORK/file1.json",
    "file2-everywhere": "C:/Users/WORK/file2.json",
    "file3": "C:/Users/WORK/file3.json"
},
"hostName": "workHost",
```

_syncLocalSettings.json at **HOME**_
```
"customFiles": {
    "file1-home": "C:/Users/HOME/file1.json",
    "file2-everywhere": "C:/Users/HOME/file2.json"
},
"hostName": "homeHost",
```

In Gist it will look like this:
|customized_sync|file1-work
|customized_sync|file1-home
|customized_sync|file2-everywhere
|customized_sync|file3

This almost works at the moment, the only issues are:
1. "file1-work" is overwritten and renamed by "file1-home" when "HOME" is uploading
2. "file3" is removed from Gist when "HOME" is uploading.

So the changes that needs to be made for this to work are:
1. The extension should looks at the Gist "file" name instead of the actual file name during upload/download
2. Add an option to control if files should/shouldn't be removed from Gist if they are unavailable in the `customFiles` setting.

Is there anything that I've overlooked or is this doable?

**Comment by shanalikhan** [comment]
Is there any possibility the scenario you have described can be achieved with GitHub Git Sync instead of Github Gist Sync, mainly because i dont want to increase more features for Gist instead Git Repo is more feasible and scalable option for many scenarios.

**Comment by depascalis** [comment]
@shanalikhan yes it would probably be possible with both of them. But, I'm not sure what you're referring to with git sync. Is it another extension or do you mean to sync to git via your extension instead, is this already possible?

**Comment by shanalikhan** [comment]
I'm planning to provide other options to Sync. Currently Settings Sync supports Gist only.
I'm planning to introduce the Git sync too. Allowing user to create repo and upload the files over there instead of Gist. 
So I think we can keep this feature for Git Sync (until someone send me PR )

**Comment by depascalis** [comment]
Sounds great. My description of the request might be unclear and hard to follow. I don't believe it's that complicated to achieve what I want, but I haven't looked at it that closely. I'll try to set off some time for this and see if I can solve it myself, I'll do a PL if so.
